mbrl/models/model.py

```diff
@@ -155,15 +172,34 @@ def update(
 
         Returns:
              (float): the numeric value of the computed loss.
-
+             (dict): any additional metadata dictionary computed by :meth:`loss`.
         """
         optimizer = cast(torch.optim.Optimizer, optimizer)
         self.train()
         optimizer.zero_grad()
-        loss = self.loss(model_in, target)
-        loss.backward()
-        optimizer.step()
-        return loss.item()
+        loss_and_maybe_meta = self.loss(model_in, target)
+        if isinstance(loss_and_maybe_meta, tuple):
+            # TODO - v0.2.0 remove this back-compatibility logic
+            loss = cast(torch.Tensor, loss_and_maybe_meta[0])
+            meta = cast(Optional[Dict[str, Any]], loss_and_maybe_meta[1])
+            loss.backward()
+
+            if meta is not None:
+                with torch.no_grad():
+                    grad_norm: torch.Tensor = 0  # type: ignore
+                    for p in list(
+                        filter(lambda p: p.grad is not None, self.parameters())
+                    ):
+                        grad_norm += p.grad.data.norm(2).item() ** 2
+                    meta["grad_norm"] = grad_norm
+            optimizer.step()
+            return loss.item(), meta
+
+        else:
+            warnings.warn(_NO_META_WARNING_MSG)
+            loss_and_maybe_meta.backward()
+            optimizer.step()
+            return loss_and_maybe_meta.item()
 
```

Comment:
IIRC, these were needed in early versions of MBRL-Lib because they received sequences of tensors instead, but they are indeed pointless now, good catch!

Reply:
Will remove in a separate PR/commit.